src/app/frontend/common/components/label/label_controller.js

```diff
@@ -0,0 +1,44 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @final
+ */
+export default class LabelController {
+  /**
+   * Constructs label controller.
+   * @ngInject
+   */
+  constructor() {
+    /** @export {!Array<Object.<string,string>>} Initialized from the scope and transformed to
+     *  correct format */
+    this.labels = this.transformLabels_();
+  }
+
+  /**
+   * Transforms labels object into array of simple key, value label objects. It is required
+   * in order to use md-chips for label display.
+   * @return {!Array<Object.<string,string>>}
+   * @private
+   */
+  transformLabels_() {
```

Comment:
This was not as easy as I thought. I've had to fix karma `ng-html2js` configuration. It was not working. There were 2 issues:

1. `stripPrefix` path was wrong. It is merged with karma `basePath` and as result it was `../../src/app/frontend/`. I've changed `basePath` to point to current folder as all our `conf.paths` have `basePath` in them.
2. `moduleName` was pointing to non existing variable. It was undefined and not loaded. I've changed it to be loaded with `ng` module so it doesn't have to be mocked manually while writing tests.

Additionally in order to make `md-chips` work without additional label transformation I've had to use workaround. There are issues reported with making `ng-repeat` and `md-chips` work.

https://github.com/angular/material/issues/3482
https://github.com/angular/material/issues/2829

Last thing is that even after fixing karma and compiling directive it didn't show any `md-chip` in html code. It looks like another issue with `md-chips` directive. I've only checked if outer loop works and x `md-chips` tags are rendered.